### PR TITLE
Add PAM stack configuration, set up test users in test hook

### PR DESCRIPTION
**Pull request by alanking**
Addresses #37 
Addresses #42

Tests passed on Ubuntu 24.04 and Rockylinux 9.

### CODE REVIEW

**Comment by alanking** [comment]
Noticed a missing test and tweaked README wording a little bit. Done with pushes now, I think.

File: README.md [hunk 1]
```diff
@@ -33,13 +33,32 @@ It is **highly recommended** that TLS/SSL be required in the server when using t
...
+
+Note: The value of this configuration is used as the `service_name` of `pam_start`. If the configured `pam_stack_name` does not exist in `/etc/pam.d` or `/etc/pam.conf`, the PAM system will fall back to its default behavior. This default behavior varies from platform to platform, so the administrator should take care that the system is properly configured. For more information about configuring PAM services, see the documentation: [https://linux.die.net/man/5/pam.conf](https://linux.die.net/man/5/pam.conf). For more information about `pam_start`, see the documentation: [https://linux.die.net/man/3/pam_start](https://linux.die.net/man/3/pam_start)
```
Comment (korydraughn) [inline README.md:42]: "of `pam_start`" sounds a little weird. Is that wording used by the man pages for PAM?

How do you feel about "to `pam_start`" or "for `pam_start`"?

Hmm, not sure I like those either. Is there other wording we can use for that?

If other feel it is fine as is, carry on. Can always tweak later if needed.

**Review by korydraughn — COMMENTED** [review]
Looks good.

File: README.md (hunk 1 above)
Comment (alanking) [reply README.md:42]: So this is the sentence in question...

> The value of this configuration is used as the `service_name` of `pam_start`.

What about this?

The value of this configuration is used as the value for the `service_name` argument for `pam_start`.

Comment (trel) [reply README.md:42]: better yes.

Comment (korydraughn) [reply README.md:42]: Agreed.

Comment (alanking) [reply README.md]: Done

**Comment by korydraughn** [comment]
I think we're ready for squashing.

**Comment by alanking** [comment]
Squashed

**Review by korydraughn — APPROVED** [review]
Pound / Merge when ready.

**Comment by alanking** [comment]
#'d, mergin

**Pull request merged**